// intl-messageformat-parser/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Type Guards
 */
function isLiteralElement(el) {
    return el.type === 0 /* literal */;
}
exports.isLiteralElement = isLiteralElement;
function isArgumentElement(el) {
    return el.type === 1 /* argument */;
}
exports.isArgumentElement = isArgumentElement;
function isNumberElement(el) {
    return el.type === 2 /* number */;
}
exports.isNumberElement = isNumberElement;
function isDateElement(el) {
    return el.type === 3 /* date */;
}
exports.isDateElement = isDateElement;
function isTimeElement(el) {
    return el.type === 4 /* time */;
}
exports.isTimeElement = isTimeElement;
function isSelectElement(el) {
    return el.type === 5 /* select */;
}
exports.isSelectElement = isSelectElement;
function isPluralElement(el) {
    return el.type === 6 /* plural */;
}
exports.isPluralElement = isPluralElement;
function createLiteralElement(value) {
    return {
        type: 0 /* literal */,
        value: value
    };
}
exports.createLiteralElement = createLiteralElement;
function createNumberElement(value, style) {
    return {
        type: 2 /* number */,
        value: value,
        style: style
    };
}
exports.createNumberElement = createNumberElement;
